-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1065c/22798/summary.html
**COMMIT**: 83b357f4dc7872fa9b679df642056459bfcb7ba2
**CMSSW**: CMSSW_12_3_X_2022-03-02-2300/slc7_amd64_gcc10
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/37123/22798/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @mmusich cms-sw/cmssw#37084
 - @Dr15Jones cms-sw/cmssw#37118

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1065c/22798/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a1065c/22798/git-merge-result


## Build

I found compilation error when building: 

<pre>>> Done generating edm plugin poisoned information
gmake[1]: *** [config/SCRAM/GMake/Makefile.rules:1852: CompilePython] Error 1
>> Plugins of all types refreshed.
gmake[1]: Target 'PostBuild' not remade because of errors.
gmake[1]: Leaving directory '/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-03-02-2300'
gmake: *** [config/SCRAM/GMake/Makefile.rules:1740: src] Error 2
gmake: Target 'all' not remade because of errors.
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
+ eval scram build outputlog '&&' '(python3' /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --ignoreWarning=Wdeprecated-declarations --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_3_X_2022-03-02-2300/tmp/slc7_amd64_gcc10/cache/log/src '||' 'true)'
++ scram build outputlog
>> Entering Package Configuration/PyReleaseValidation
</pre>

--- a/Configuration/PyReleaseValidation/python/relval_gpu.py
+++ b/Configuration/PyReleaseValidation/python/relval_gpu.py
@@ -14,39 +14,39 @@ from Configuration.PyReleaseValidation.relval_upgrade import workflows as _upgra
 #just define all of them
 
 #WFs to run in IB:
-# mc 2018   (Patatrack pixel-only quadruplets: ZMM - on GPU, both CPU and GPU, auto)
-#           (Patatrack pixel-only triplets: ZMM - on GPU, both CPU and GPU, auto)
-#           (Patatrack pixel-only quadruplets: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack pixel-only triplets: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack ECAL-only: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack HCAL-only: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack full reco with pixel quadruplets: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack full reco with pixel triplets: TTbar - on GPU, both CPU and GPU, auto)
-# mc 2021   (Patatrack pixel-only quadruplets: ZMM - on GPU, both CPU and GPU, auto)
-#           (Patatrack pixel-only triplets: ZMM - on GPU, both CPU and GPU, auto)
-#           (Patatrack pixel-only quadruplets: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack pixel-only triplets: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack ECAL-only: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack HCAL-only: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack full reco with pixel quadruplets: TTbar - on GPU, both CPU and GPU, auto)
-#           (Patatrack full reco with pixel triplets: TTbar - on GPU, both CPU and GPU, auto)
+# mc 2018   (Patatrack pixel-only quadruplets: ZMM - on GPU, both CPU and GPU, profiling)
+#           (Patatrack pixel-only triplets: ZMM - on GPU, both CPU and GPU, profiling)
+#           (Patatrack pixel-only quadruplets: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack pixel-only triplets: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack ECAL-only: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack HCAL-only: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack full reco with pixel quadruplets: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack full reco with pixel triplets: TTbar - on GPU, both CPU and GPU, profiling)
+# mc 2021   (Patatrack pixel-only quadruplets: ZMM - on GPU, both CPU and GPU, profiling)
+#           (Patatrack pixel-only triplets: ZMM - on GPU, both CPU and GPU, profiling)
+#           (Patatrack pixel-only quadruplets: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack pixel-only triplets: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack ECAL-only: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack HCAL-only: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack full reco with pixel quadruplets: TTbar - on GPU, both CPU and GPU, profiling)
+#           (Patatrack full reco with pixel triplets: TTbar - on GPU, both CPU and GPU, profiling)
 numWFIB = [
-           10842.502, # 10842.503,10842.504,
-           10842.506, # 10842.507,10842.508,
-           10824.502, # 10824.503,10824.504,
-           10824.506, # 10824.507,10824.508,
-           10824.512, # 10824.513,10824.514,
-           10824.522, # 10824.523,10824.524,
-           10824.592, # 10824.593,10824.594,
-           10824.596, # 10824.597,10824.598,
-           11650.502, # 11650.503,11650.504,
-           11650.506, # 11650.507,11650.508,
-           11634.502, # 11634.503,11634.504,
-           11634.506, # 11634.507,11634.508,
-           11634.512, # 11634.513,11634.514,
-           11634.522, # 11634.523,11634.524
-           11634.592, # 11634.593,11634.594,
-           11634.596, # 11634.597,11634.598,
+           10842.502, # 10842.503, 10842.504,
+           10842.506, # 10842.507, 10842.508,
+           10824.502, # 10824.503, 10824.504,
+           10824.506, # 10824.507, 10824.508,
+           10824.512, 10824.513, # 10824.514,
+           10824.522, # 10824.523, 10824.524,
+           10824.592, # 10824.593, 10824.594,
+           10824.596, # 10824.597, 10824.598,
+           11650.502, # 11650.503, 11650.504,
+           11650.506, # 11650.507, 11650.508,
+           11634.502, # 11634.503, 11634.504,
+           11634.506, # 11634.507, 11634.508,
+           11634.512, 11634.513, # 11634.514,
+           11634.522, # 11634.523, 11634.524,
+           11634.592, # 11634.593, 11634.594,
+           11634.596, # 11634.597, 11634.598,
         ]
 for numWF in numWFIB:
     if not numWF in _upgrade_workflows: continue

--- a/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py
+++ b/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py
@@ -682,12 +682,12 @@ upgradeWFs['PatatrackECALOnlyGPU'] = PatatrackWorkflow(
 upgradeWFs['PatatrackECALOnlyGPUValidation'] = PatatrackWorkflow(
     digi = {
         '--accelerators': 'gpu-nvidia',
-        '--procModifiers': 'gpu,gpuValidationEcal'
+        '--procModifiers': 'gpu'
     },
     reco = {
         '-s': 'RAW2DIGI:RawToDigi_ecalOnly,RECO:reconstruction_ecalOnly,VALIDATION:@ecalOnlyValidation,DQM:@ecalOnly',
         '--accelerators': 'gpu-nvidia',
-        '--procModifiers': 'gpu,gpuValidationEcal'
+        '--procModifiers': 'gpuValidation'
     },
     harvest = {
         '-s': 'HARVESTING:@ecalOnlyValidation+@ecal'

--- a/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py
+++ b/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py
@@ -681,12 +681,12 @@ upgradeWFs['PatatrackECALOnlyGPU'] = PatatrackWorkflow(
 
 upgradeWFs['PatatrackECALOnlyGPUValidation'] = PatatrackWorkflow(
     digi = {
-        '--accelerators': 'gpu-nvidia'
+        '--accelerators': 'gpu-nvidia',
         '--procModifiers': 'gpu,gpuValidationEcal'
     },
     reco = {
         '-s': 'RAW2DIGI:RawToDigi_ecalOnly,RECO:reconstruction_ecalOnly,VALIDATION:@ecalOnlyValidation,DQM:@ecalOnly',
-        '--accelerators': 'gpu-nvidia'
+        '--accelerators': 'gpu-nvidia',
         '--procModifiers': 'gpu,gpuValidationEcal'
     },
     harvest = {

